src/ProjectTemplates/test/BlazorWasmTemplateTest.cs

```diff
@@ -102,16 +174,25 @@ protected async Task BuildAndRunTest(string appName, Project project)
             if (BrowserFixture.IsHostAutomationSupported())
             {
                 aspNetProcess.VisitInBrowser(Browser);
-                TestBasicNavigation(appName);
+                TestBasicNavigation(appName, usesAuth);
             }
             else
             {
                 BrowserFixture.EnforceSupportedConfigurations();
             }
         }
 
-        private void TestBasicNavigation(string appName)
+        private void TestBasicNavigation(string appName, bool usesAuth = false)
         {
+            // Start fresh always
+            if (usesAuth)
+            {
+                Browser.ExecuteJavaScript("sessionStorage.clear()");
+                Browser.ExecuteJavaScript("localStorage.clear()");
```

Comment:
As an alternative to this, you could start up the browser with a unique isolation string: https://github.com/dotnet/aspnetcore/pull/18859/files#diff-aa4dc111acb9971dc780b627d9ac9f8aR37

That gives you a clean browser state that has never had anything in `sessionStorage`/`localStorage`.

What you've done here presumably also works though so I'm not saying it necessarily must be changed!